openmdao/core/explicitcomponent.py

```diff
@@ -178,7 +179,11 @@ def _apply_nonlinear(self):
 
                 self._inputs.read_only = True
                 try:
-                    self.compute(self._inputs, outputs)
+                    if self._discrete_inputs or self._discrete_outputs:
```

Comment:
Isn't there a way to pack the discrete vars into the `inputs` and `outputs` dict-like objects that the user sees, so that the api for the call doesn't need additional arguments?

Reply:
We would have to make some sort of dict wrapper to contain both discrete and non-discrete.  Justin and I discussed this as an option but the decision was made to go with just having a different arg list for components that actually use discrete vars.